spicy/toolchain/src/compiler/codegen/unit-builder.cc

```diff
@@ -219,6 +221,23 @@ UnqualifiedType* CodeGen::compileUnit(type::Unit* unit, bool declare_only) {
     add_hook("0x25_undelivered", {builder()->parameter("seq", builder()->typeUnsignedInteger(64)),
                                   builder()->parameter("data", builder()->typeBytes())});
 
+    auto attr_uses_stream = builder()->attribute("&needed-by-feature", builder()->stringLiteral("uses_stream"));
+    auto stream =
+        builder()->declarationField(ID("__stream"),
+                                    builder()->qualifiedType(builder()->typeWeakReference(
+                                                                 builder()->qualifiedType(builder()->typeStream(),
+                                                                                          hilti::Constness::Const)),
+                                                             hilti::Constness::Const),
+                                    builder()->attributeSet({builder()->attribute("&internal"), attr_uses_stream}));
+
+    v.addField(stream);
+
+    auto attr_sync_advance = builder()->attributeSet(
+        {builder()->attribute("&needed-by-feature", builder()->stringLiteral("uses_sync_advance"))});
+
+    add_hook("0x25_sync_advance", {builder()->parameter("offset", builder()->typeUnsignedInteger(64))},
+             attr_sync_advance);
```

Comment:
Not sure we need this optimization and associated feature constant at this point, but maybe worth leaving it in since we have it.

Reply:
Not sure I follow. The optimization is to avoid calling the hook if we don't have an implementation (along with pre/post hook stuff). Do you mean something else?

Reply:
You are right, I got slightly confused. What we already do today is remove hooks which are not implemented, but the situation here is different since we want to call the hook depending on a feature constant. Resolving.